### PR TITLE
Add local emulator backend

### LINKED ISSUE #65: `IBMQEmulatorBackend` variant that runs locally
Since adoption of qiskit runtime the `IBMQEmulatorBackend` became a runtime instance of the `ibmq_qasm_simulator` backend (hosted by IBM). It would still be useful to have a local emulator based on `AerBackend` using an imported noise model.

### PR DESCRIPTION
**Pull request by cqc-alec**
# Description

Add a new backend `IBMQLocalEmulatorBackend` which uses `AerBackend` with the noise model and characteristics of the real device. This is similar to `IBMQEmulatorBackend` except that it does all emulation locally. The only outward difference is that it does not support contextual optimization (because `AerBackend` doesn't).

# Related issues

Closes #65 .

# Checklist

- [x] I have performed a self-review of my code.
- [x] I have commented hard-to-understand parts of my code.
- [x] I have made corresponding changes to the public API documentation.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have updated the changelog with any user-facing changes.

### CODE REVIEW

**Comment by CalMacCQ** [comment]
Would be good to add the `IBMQLocalEmulatorBackend` to the [table of available backends ](https://github.com/CQCL/pytket-qiskit/blob/develop/docs/intro.txt#L30).

**Comment by cqc-alec** [comment]
> Would be good to add the `IBMQLocalEmulatorBackend` to the [table of available backends ](https://github.com/CQCL/pytket-qiskit/blob/develop/docs/intro.txt#L30).

I think I did that as part of this PR, or did I miss something?

**Comment by CalMacCQ** [comment]
> > Would be good to add the `IBMQLocalEmulatorBackend` to the [table of available backends ](https://github.com/CQCL/pytket-qiskit/blob/develop/docs/intro.txt#L30).
> 
> I think I did that as part of this PR, or did I miss something?

Ah yes you did but I changed the formatting of the table in #265 . Sorry about that.

**Comment by cqc-alec** [comment]
> > > Would be good to add the `IBMQLocalEmulatorBackend` to the [table of available backends ](https://github.com/CQCL/pytket-qiskit/blob/develop/docs/intro.txt#L30).
> > 
> > 
> > I think I did that as part of this PR, or did I miss something?
> 
> Ah yes you did but I changed the formatting of the table in #265 . Sorry about that.

No problem, I'll merge and fix it up.

File: pytket/extensions/qiskit/backends/ibmq_emulator.py [hunk 1]
```diff
@@ -54,10 +54,10 @@
...
 
 class IBMQEmulatorBackend(Backend):
-    """A backend which uses the ibmq_qasm_simulator to emulate the behaviour of
-    IBMQBackend. Performs the same compilation and predicate checks as IBMQBackend.
-    Requires a valid IBMQ account.
-
+    """A Backend which uses the ibmq_qasm_simulator to emulate the behaviour of
+    IBMQBackend. Identical to :py:class:`IBMQBackend` except there is no `monitor`
+    parameter. Performs the same compilation and predicate checks as IBMQBackend.
+    Requires a valid IBM account.
```
Comment (cqc-melf) [inline pytket/extensions/qiskit/backends/ibmq_emulator.py:60]: What is the difference of an IBM and an IBMQ account? Should we update this everywhere?

File: pytket/extensions/qiskit/backends/ibmq_emulator.py [hunk 2]
```diff
@@ -54,7 +54,7 @@
 
 
 class IBMQEmulatorBackend(Backend):
-    """A Backend which uses the AerBackend simulator to emulate the behaviour of
+    """A Backend which uses the ibmq_qasm_simulator to emulate the behaviour of
     IBMQBackend. Identical to :py:class:`IBMQBackend` except there is no `monitor`
     parameter. Performs the same compilation and predicate checks as IBMQBackend.
     Requires a valid IBM account.
```
Comment (cqc-melf) [inline pytket/extensions/qiskit/backends/ibmq_emulator.py:60]:
```suggestion
    Requires a valid IBMQ account.

**Review by cqc-melf — APPROVED** [review]

**Pull request merged**